### DISCUSSION TITLE
Questions About the Current State of the Project

### DISCUSSION THREAD

**Discussion by Jakob-PB** [MirageNet/Mirage#762] [Q&A]
Hello devs! Mirage looks like a really interesting project that upgrades Mirror in all of the ways I would like it to. In particular I appreciate that it supports disabling domain reload in the editor which is a major reason why I would rather use this library than Mirror.

What I was wondering is where do you see the current state of the project? It seems like a fast moving target with a lot of major changes on the roadmap and frequent updates. How much do you expect the API to change from here on out? Mirror is pretty stable which is highly appealing to me compared to say Unity's new MLAPI. Do you see Mirage like Mirror with a couple of nice changes patched on top or is this a major fork with none of the same stability guarantees coming along for the ride? It seems like you don't pull anything from upstream which is concerning considering the benefits of such a large project and community catching bugs. My major worry here is that Mirage is too much of a divergence, which makes it much more of a gamble for a dev like me to try and use.

Thanks for reading!

**Comment by Lymdun** [comment]
Hello and thanks for your interest!

About the coming API changes, the main and biggest changes are the transport layer rework (#749) and the new AOI system (#743) which will provide a cleaner API and way better performance for the future. Although, it would be more correct to see Mirage as a major fork and even a different networking library. We simplified/cleaned/rewrote lot of code at the expense of backwards compatibility and basically rewrote the architecture, which brought bugfixes as well, and added new features. As an example, before the repository left its "fork status", the amount of commits on top of Mirror for this repo was thousands, which is why we don't pull anything from upstream ; code is now very different from Mirror, most known bugs got fixed along the road, or are currently being fixed.
You're right, we don't have the benefits of having a big community to catch bugs, but we have a fairly advanced test suite which covers more than 80% of Mirage's code (!), so we believe in Mirage having a low amount of bugs.

Side-note: you can see most of API changes here: https://miragenet.github.io/Mirage/Articles/Guides/MirrorMigration.html

If you need any help or have any more questions, please feel free to ask them here or to join our Discord server!

**Reply by Jakob-PB** [reply]
Thank you for the detailed response, that really cleared things up!

By the way, I had no idea you guys had a documentation website, and I looked around here quite a bit! I totally missed that tiny little documentation button. You should really consider putting a link to it under the about section instead of linking to the Unity forum thread for Mirror.
